apps/offline-jobs/src/crons/cache-warmer/cache-warmer.service.ts

```diff
@@ -11,44 +12,65 @@ export class CacheWarmerService {
 
     await this.cacheService.set(CacheInfo.PairsMetadata().key, pairsMetadata, CacheInfo.PairsMetadata().ttl);
     this.clientProxy.emit('deleteCacheKeys', [CacheInfo.PairsMetadata().key]);
   }
 
-  @Cron(CronExpression.EVERY_HOUR)
-  @Lock({ name: 'warmTokens', verbose: true })
+  @Cron(CronExpression.EVERY_10_MINUTES)
+  @Lock({ name: 'Warm OneDex pairs metadata', verbose: true })
+  async warmOneDexPairsMetadata() {
+    const pairsMetadata = await this.oneDexService.getPairsMetadataRaw();
+
+    await this.cacheService.set(CacheInfo.OneDexPairsMetadata().key, pairsMetadata, CacheInfo.OneDexPairsMetadata().ttl);
+    this.clientProxy.emit('deleteCacheKeys', [CacheInfo.OneDexPairsMetadata().key]);
+  }
+
+  @Cron(CronExpression.EVERY_2ND_HOUR)
+  @Lock({ name: 'Warm Tokens', verbose: true })
   async warmTokens() {
-    const pairsMetadata = await this.xExchangeService.getPairsMetadata();
-    const tokens = pairsMetadata.map(pair => [pair.firstTokenId, pair.secondTokenId]).flat().distinct();
+    const xExchangePairsMetadata = await this.xExchangeService.getPairsMetadata();
+    const oneDexPairsMetadata = await this.oneDexService.getPairsMetadata();
 
-    const cacheKeys: string[] = [];
-    for (const tokenId of tokens) {
-      try {
-        const token = await this.multiversXApiService.getTokenRaw(tokenId);
+    const xExchangeTokens = xExchangePairsMetadata.map(pair => [pair.firstTokenId, pair.secondTokenId]).flat();
+    const oneDexTokens = oneDexPairsMetadata.map(pair => [pair.first_token_id, pair.second_token_id]).flat();
 
-        await this.cacheService.set(CacheInfo.Token(tokenId).key, token, CacheInfo.Token(tokenId).ttl);
-        cacheKeys.push(CacheInfo.Token(tokenId).key);
-      } catch (error: any) {
-        this.logger.error(`Failed to get token with identifier: ${tokenId}`);
-        this.logger.error(error);
+    const tokens = [...xExchangeTokens, ...oneDexTokens].distinct();
+    const tokenBatches = BatchUtils.splitArrayIntoChunks(tokens, 25);
+
+    for (const batch of tokenBatches) {
+      const cacheKeys: string[] = [];
+
+      for (const tokenId of batch) {
+        try {
+          const token = await this.multiversXApiService.getTokenRaw(tokenId);
+
+          await this.cacheService.set(CacheInfo.Token(tokenId).key, token, CacheInfo.Token(tokenId).ttl);
+          cacheKeys.push(CacheInfo.Token(tokenId).key);
+        } catch (error: any) {
+          this.logger.error(`Failed to get token with identifier: ${tokenId}`);
+          this.logger.error(error);
+        }
       }
-    }
 
-    this.clientProxy.emit('deleteCacheKeys', cacheKeys);
+      this.clientProxy.emit('deleteCacheKeys', cacheKeys);
+
+      await new Promise(r => setTimeout(r, 5000));
+    }
   }
 
-  @Cron(CronExpression.EVERY_10_MINUTES)
-  @Lock({ name: 'warmPairFees', verbose: true })
+  @Cron(CronExpression.EVERY_30_MINUTES)
+  @Lock({ name: 'Warm xExchange pair fees', verbose: true })
```

Comment:
shouldn`t we do the same for onedex ?

Reply:
the pairs from onedex already come with the fee in the vm-query response. On xexchange a separate query is made to get this info